lib/Backend/TTNN/TTNNWrapperLib.cpp

```diff
@@ -0,0 +1,203 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstddef>
+#include <llvm/Support/Casting.h>
+#include <mlir/IR/AttrTypeSubElements.h>
+#include <stdexcept>
+
+#include "TTNNWrapper.hpp"
+#include "TTNNWrapperLib_Impl.hpp"
+#include "impl/buffers/buffer.hpp"
+#include "ttmlir/Dialect/TT/IR/TTOpsTypes.h"
+#include "ttnn/tensor/types.hpp"
+
+namespace mlir::tt::backend::ttnn {
+
+// alias to a common backend types
+using be_DataType = ::tt::tt_metal::DataType;
+using be_Layout = ::tt::tt_metal::Layout;
+using be_CoreRange = ::tt::tt_metal::CoreRange;
+using be_CoreRangeSet = ::tt::tt_metal::CoreRangeSet;
+using be_CoreCoord = ::tt::tt_metal::CoreCoord;
+using be_ShardSpec = ::tt::tt_metal::ShardSpec;
+using be_ShardOrientation = ::tt::tt_metal::ShardOrientation;
+using be_TensorMemoryLayout = ::tt::tt_metal::TensorMemoryLayout;
+using be_MemoryConfig = ::tt::tt_metal::MemoryConfig;
```

Comment:
I don't think the `be_` is required, we should use namespaces for this kind of thing and you'd already namespace this.